compiler/qsc_doc_gen/src/generate_docs.rs

```diff
@@ -316,102 +449,213 @@ Fully qualified name: {fqn}
+
+    // This inserts the top index into the table of contents.
+    // The file itself is inserted in the npm/generate_docs.js file.
+    let top_index = "- items:
+  name: Overview
+  uid: Microsoft.Quantum.apiref-toc";
+
+    let mut table = map
+        .iter_mut()
+        .map(|(namespace, items)| {
+            let items_str = items
+                .iter()
+                .map(|item| format!("  - {{name: {}, uid: {}}}", item.name, item.uid))
+                .collect::<Vec<String>>()
+                .join("\n");
+            let content =
+                format!("- items:\n{items_str}\n  name: {namespace}\n  uid: Qdk.{namespace}");
+            (namespace, content)
+        })
+        .collect::<Vec<_>>();
+
+    table.sort_unstable_by_key(|(n, _)| {
+        // Ensures that the Microsoft.Quantum.Unstable namespaces are listed last.
+        if n.starts_with("Microsoft.Quantum.Unstable") {
+            format!("1{n}")
+        } else {
+            format!("0{n}")
+        }
+    });
+    let table = table
+        .into_iter()
+        .map(|(_, c)| c)
+        .collect::<Vec<_>>()
+        .join("\n");
+    let content = format!("{header}\n{top_index}\n{table}");
+
+    let file_name: Arc<str> = Arc::from("toc.yml");
+    let file_metadata: Arc<str> = Arc::from("");
+    let file_content: Arc<str> = Arc::from(content.as_str());
+    files.push((file_name, file_metadata, file_content));
 }
 
-enum MetadataKind {
-    Function,
-    Operation,
-    Udt,
-    Export,
+fn get_namespace(package: &Package, item: &Item) -> Option<Rc<str>> {
+    match item.parent {
+        Some(local_id) => {
+            let parent = package
+                .items
+                .get(local_id)
+                .expect("Could not resolve parent item id");
+            match &parent.kind {
+                ItemKind::Namespace(name, _) => {
+                    if name.starts_with("QIR") {
+                        None // We ignore "QIR" namespaces
+                    } else {
+                        Some(name.name())
+                    }
+                }
+                _ => None,
+            }
+        }
+        None => None,
+    }
 }
 
-impl Display for MetadataKind {
-    fn fmt(&self, f: &mut Formatter<'_>) -> Result {
-        let s = match &self {
-            MetadataKind::Function => "function",
-            MetadataKind::Operation => "operation",
-            MetadataKind::Udt => "user defined type",
-            MetadataKind::Export => "exported item",
-        };
-        write!(f, "{s}")
+// Recursively resolves export items until it can find the root definition.
+// Returns the package and item of the root definition for an item.
+// If given the root definition, it will return the same item.
+fn resolve_export<'a>(
+    default_package_id: PackageId,
+    package: &'a Package,
+    include_internals: bool,
+    item: &'a Item,
+    display: &'a CodeDisplay,
+) -> Option<(&'a Package, &'a Item)> {
+    // Filter items
```

Comment:
perhaps specify what exactly is being filtered out -- the code kind of says it, but since this comment is already here..might as well make it a full sentence 😸

Reply:
expanded the comment to be more descriptive